kernel/constr.ml

```diff
@@ -1173,13 +1148,10 @@ let hasheq t1 t2 =
       && array_eqeq lna1 lna2
       && array_eqeq tl1 tl2
       && array_eqeq bl1 bl2
-    | Int i1, Int i2 -> i1 == i2
-    | Float f1, Float f2 -> Float64.equal f1 f2
-    | Array(u1,t1,def1,ty1), Array(u2,t2,def2,ty2) ->
-      u1 == u2 && def1 == def2 && ty1 == ty2 && array_eqeq t1 t2
+    | PVal v1, PVal v2 -> v1 == v2
```

Comment:
There is obviously a change of semantics here, not sure how serious it is if the function is only playing a heuristic role for hashing. It might be safer to either avoid the change or isolate it in a separate commit.

Reply:
I don't think there should be a change of semantics here, as places that used to preserve physical equality as much as possible still do (at least, that is what I intended).

Reply:
Well, this is at least assuming that Float64.equal and == are the same thing, which is wrong.